### DISCUSSION TITLE
option to display progress in percentage only

### DISCUSSION THREAD

**Discussion by adrelanos** [curl/curl#12306] [Ideas]
So we don't need complex scripts such as [curl-progress](https://gist.github.com/sstephenson/4587282) could you please add feature to show the progress? Syntax ideas:

* `curl --progress-file /path/to/progress-file`
* `curl --write-out '%{progress_download}` [1]

Percentage only. Nothing else. Use case: usage of curl in scripts.

----

[1] This would require `--write-out` to be continuously updated.

**Comment by jay** [comment]
Why isn't `curl -# -O ...` sufficient?

**Comment by dfandrich** [comment]
See also #10372.

**Comment by adrelanos** [comment]
> Why isn't `curl -# -O ...` sufficient?

Difficult to parse with a script. Could be related to buffering.

    curl -# url --output /tmp/x 2>&1 | echo

I got no output.

Also awk / sed are awful, make the code more complicated to read, add additional attack surface and depend on the output of the progress bar to not change.

Also the necessity of `2>&1` makes things more complicated.
